framework/test/jest/unit/specs/application/node/forger/forger.spec.js

```diff
@@ -878,6 +1048,200 @@ describe('forger', () => {
+				// Arrange
+				const targetDelegate = genesisDelegates.delegates[0];
+				jest
+					.spyOn(forger, 'getDelegateKeypairForCurrentSlot')
+					.mockResolvedValue(targetDelegate);
+				forgeModule.config.forging.delegates = genesisDelegates.delegates;
+				when(forgeModule.storage.entities.ForgerInfo.getKey)
+					.calledWith(FORGER_INFO_KEY_USED_HASH_ONION)
+					.mockResolvedValue(
+						JSON.stringify([
+							{
+								count: 5,
+								height: 9,
+								address: getAddressFromPublicKey(targetDelegate.publicKey),
+							},
+							{
+								count: 6,
+								height: 12,
+								address: getAddressFromPublicKey(targetDelegate.publicKey),
+							},
+						]),
+					);
+
+				// Act
+				await forgeModule.forge();
+				// Assert
+				expect(
+					forgeModule.storage.entities.ForgerInfo.setKey,
+				).toHaveBeenCalledWith(
+					FORGER_INFO_KEY_USED_HASH_ONION,
+					JSON.stringify([
+						{
+							count: 5,
+							height: 9,
+							address: getAddressFromPublicKey(targetDelegate.publicKey),
+						},
+						{
+							count: 6,
+							height: 12,
+							address: getAddressFromPublicKey(targetDelegate.publicKey),
+						},
+						{
+							count: 7,
+							address: getAddressFromPublicKey(targetDelegate.publicKey),
+							height: lastBlock.height + 1,
+						},
+					]),
+				);
+			});
+
+			it('should overwrite the used hash onion when forgin the same height', async () => {
+				const targetDelegate = genesisDelegates.delegates[0];
+				jest
+					.spyOn(forger, 'getDelegateKeypairForCurrentSlot')
+					.mockResolvedValue(targetDelegate);
+				forgeModule.config.forging.delegates = genesisDelegates.delegates;
+				when(forgeModule.storage.entities.ForgerInfo.getKey)
+					.calledWith(FORGER_INFO_KEY_USED_HASH_ONION)
+					.mockResolvedValue(
+						JSON.stringify([
+							{
+								count: 5,
+								height: 9,
+								address: getAddressFromPublicKey(targetDelegate.publicKey),
+							},
+							{
+								count: 6,
+								height: 12,
+								address: getAddressFromPublicKey(targetDelegate.publicKey),
+							},
+							{
+								count: 7,
+								height: lastBlock.height + 1,
+								address: getAddressFromPublicKey(targetDelegate.publicKey),
+							},
+						]),
+					);
+
+				// Act
+				await forgeModule.forge();
+				// Assert
+				expect(
+					forgeModule.storage.entities.ForgerInfo.setKey,
+				).toHaveBeenCalledWith(
+					FORGER_INFO_KEY_USED_HASH_ONION,
+					JSON.stringify([
+						{
+							count: 5,
+							height: 9,
+							address: getAddressFromPublicKey(targetDelegate.publicKey),
+						},
+						{
+							count: 6,
+							height: 12,
+							address: getAddressFromPublicKey(targetDelegate.publicKey),
+						},
+						{
+							count: 7,
+							address: getAddressFromPublicKey(targetDelegate.publicKey),
+							height: lastBlock.height + 1,
+						},
+					]),
+				);
+			});
+
+			it('should remove all used hash onions before finality height', async () => {
```

Comment:
I dont see anywhere the assertion for removal, or am I missing something?

Reply:
`count 5` is removed in the expect field